educator/src/Dscp/Educator/Web/Educator/Auth.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | Necessary types and implementation for Educator authenthication
+module Dscp.Educator.Web.Educator.Auth
+       ( EducatorPublicKey (..)
+       , EducatorAuth
+       ) where
+
+import Crypto.JOSE.JWK (KeyMaterial (..), KeyOp (..), OKPKeyParameters (..), jwkKeyOps)
+import Crypto.JWT (fromKeyMaterial)
+import Servant.Auth.Server (AuthCheck (..), defaultJWTSettings, jwtAuthCheck)
+import Servant.Auth.Server.Internal.Class (AuthArgs (..), IsAuth (..))
+
+import Dscp.Crypto (AbstractPK (..), PublicKey)
+import Dscp.Educator.Web.Auth (AuthData)
+
+---------------------------------------------------------------------------
+-- Data types
+---------------------------------------------------------------------------
+
+-- | Custom authentication type for auth-servant
+data EducatorAuth
+
+-- | Type that holds Educator's public key
+newtype EducatorPublicKey = EducatorPublicKey PublicKey
+
+instance IsAuth EducatorAuth (AuthData, ()) where
```

Comment:
Again -- this looks kind of weird. I'd expect `IsAuth EducatorAuth EducatorAuthData` or something, but not a `()`.